models/Monte_Carlo.py

```diff
@@ -0,0 +1,166 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from arc import *
+import scipy.constants as const
+from scipy.integrate import odeint
+
+def Monte_Carlo_iHO_3d(atom, t, omega_H, omega_L, T, N_ensemble, traping_laser, D2_line):
+    
+    def wavelength2frequency (wavelength):
+        from scipy.constants import c as c_c
+        return c_c/(wavelength)
+    
+    def iHO_a_3d (x, y, z, omega_H, omega_L):
+        ax = (omega_H**2)*x
+        ay = (omega_H**2)*y
+        g = 9.8 #m/s^2 gravity
+        az = (omega_L**2)*z - g
+        return ax, ay, az
+    
+    def newton_3d(f, t, omega_H, omega_L):
+        x, vx, y, vy, z, vz = f
+        ax, ay, az = iHO_a_3d(x, y, z, omega_H, omega_L)
+        dfdt = [vx, ax, vy, ay, vz, az]
+        return dfdt
+
+    m = atom.mass
+    delta_x = np.sqrt(const.k*T/(m*omega_H**2))
+    delta_vx = np.sqrt(const.k*T/m)
+    delta_y = np.sqrt(const.k*T/(m*omega_H**2))
+    delta_vy = np.sqrt(const.k*T/m)
+    delta_z = np.sqrt(const.k*T/(m*omega_L**2))
+    delta_vz = np.sqrt(const.k*T/m)
+    x_ensemble = np.random.normal(0, delta_x, N_ensemble)
+    y_ensemble = np.random.normal(0, delta_y, N_ensemble)
+    z_ensemble = np.random.normal(0, delta_z, N_ensemble)
+    vx_ensemble = np.random.normal(0, delta_vx, N_ensemble)
+    vy_ensemble = np.random.normal(0, delta_vy, N_ensemble)
+    vz_ensemble = np.random.normal(0, delta_vz, N_ensemble)
+
+
+    v = wavelength2frequency(traping_laser)
+    v0 = wavelength2frequency(D2_line)
+    alpha = abs((v + v0)*(v-v0)/v**2)
+    print(f"alpha = {alpha}")
+
+    omega_iHO_H = np.sqrt(alpha)*omega_H
+    omega_iHO_L = np.sqrt(alpha)*omega_L
+
+    x_ensemble_t = []
+    y_ensemble_t = []
+    z_ensemble_t = []
+    vx_ensemble_t = []
+    vy_ensemble_t = []
+    vz_ensemble_t = []
+
+    for x, vx, y, vy, z, vz in zip(x_ensemble, vx_ensemble, y_ensemble, vy_ensemble, z_ensemble, vz_ensemble):
+        f0 = [x, vx, y, vy, z, vz]
+        sol = odeint(newton_3d, f0, t, args = (omega_iHO_H, omega_iHO_L,))
```

Comment:
Also -- I think that the equations of motion here are integrable, so you could just write $x(t) = x(0) \cosh \omega_x t + x'(0) \omega_x^{-1} \sinh \omega_x t$